packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -83,7 +83,17 @@ export default class Proxy {
     return id
   }
 
-  unregisterProxy(id) {
+  async unregisterProxy(id) {
+    const { vmUuid } = await this._getProxy(id)
+    if (vmUuid !== undefined) {
+      this._app
+        .unbindLicense({
```

Comment:
Shouldn't we make sure that we were able to forget it before unbinding?

Reply:
Make sense :+1:

Reply:
I prefer an unbound (unusable proxy) proxy to a license bound to an non-existing proxy.

Reply:
Hmm, my comment does not really make sense if we are not waiting for `unbindLicense`.

I agree with @pdonias then.